### ISSUE TITLE
Node shuts down without error after a few hours

### ISSUE THREAD

**Issue by prestwich** [celo-org/celo-blockchain#573]
### Expected Behavior

Validator should continue to run until shut off.

### Current Behavior

Validator shuts down without error message a few hours after launching. No visible error message from the log. Machine not seem to be OOM, so unsure if this is related to #532.

Copied the last few lines of log output. The previous few hours alternated blocks of these 2 log messages. Is having so many old messages normal behavior?

[james-celo-crash.log](https://github.com/celo-org/celo-blockchain/files/3816501/james-celo-crash.log)

**Comment by nategraf** [comment]
The `Invalid round change certificate with preprepare.` is interesting. @trianglesphere, do you have any ideas what that might mean?

**Comment by prestwich** [comment]
let me know if there are additional logs or info I can provide. Gonna stop running the validator as it does this consistently and it's a pain to reboot and unlock everything :)

**Comment by trianglesphere** [comment]
It doesn't seem to be #532. Having that many old messages is a little odd, that occurs before the different nodes can re-sync on their round/seq.

What's the setup for this validator? That is are you running it on betanet or on your own testnet?

**Comment by prestwich** [comment]
Running on alfajores. Followed the setup guide for a validator. Only deviated to add a third account to play around with other things

**Comment by nategraf** [comment]
I was running a validator overnight and wasn't able to reproduce this yet :/

Looking over your logs again I still don't see anything that explains the crash.

Assuming OOM is still possible you could check your system logs for signs that it killed the process.`dmesg | egrep -i 'killed process'` will do the trick if it is being killed by the system. Also, if you run `docker ps -a` you should be able to see the exit code for the container, which will be `137` is it died by OOM.

If not that, it could be helpful to increase the logging verbosity to `6` and get some particularly detailed logs.

**Comment by prestwich** [comment]
`docker ps -a` shows `137`, so the crash is likely OOM

**Comment by nategraf** [comment]
Ok, that makes sense. It seems most likely that it is related to https://github.com/celo-org/celo-blockchain/issues/532 and we are actively working on that to get it fixed. If you are inclined to do so, you could get a `pprof` dump like @tkporter did for #532 and confirm that memory is leaking in the same place.

**Comment by prestwich** [comment]
if I have time this week I'll try to `pprof`

**Comment by tkporter** [comment]
Hey @prestwich!

Just to make it easier on you this is how I got my pprof content:

1. Ran the celo-blockchain (geth) client with `--pprof --memprofilerate=524288` and let it bake for a while
2. `curl`'d http://localhost:6060/debug/pprof/heap into a file (if you're running in a container you'll have to either exec into the container or expose port 6060)
3. cd into my clone of `celo-blockchain` at the commit alfajores is currently deployed off: https://github.com/celo-org/celo-blockchain/commit/9575a01c12438e1adc6bb610c891cb5066aaa7aa
4. After running `make all`, ran `go tool pprof build/bin/geth <PATH_TO_HEAP_FILE>`

Most curious about these pprof cmds:

1. `top10`
2. `list discv5.\(\*Network\)\.lookup`
